### ISSUE TITLE
softmax_cross_entropy_with_logits - wrong backprop for this operation

### ISSUE THREAD

**Issue by berryweinst** [tensorflow/tensorflow#46672]
Hi,

I've noticed that the arithmetic for the backward operation of softmax_cross_entropy_with_logits is lacking of an additional multiplier. As can be seen in the following Xent kernel, that the operation is using, the formula for the log_cross_entropy is:
**_target - logits_**
while it should be:
_target*(1 - logits)_
When the target is dense, and doesn't sum up to exactly one, the answer is wrong. 

https://github.com/tensorflow/tensorflow/blob/dec8e0b11f4f87693b67e125e67dfbc68d26c205/tensorflow/core/kernels/xent_op.h#L118

Here is a numpy reference code for what I expect the backward should be:
```
## gradOutput is actually the target labels 
def referenceLogSoftMaxBackward(logSoftMaxOutput, gradOutput):
    '''
    accreal sum = 0;
    for (d = 0; d < dim; d++)
      sum += gradOutput_data[d];

    for (d = 0; d < dim; d++)
      gradInput_data[d] = gradOutput_data[d] - exp(output_data[d])*sum;
    '''
    shape = gradOutput.shape
    dims_to_normalize_tuple = tuple(np.arange(len(shape))[1:])

    # Sum reduction over all dimensions except for batch
    sum = np.sum(gradOutput, axis=dims_to_normalize_tuple, keepdims=True)

    # Compound operation:
    # Elementwise exponential and then multiplication by broadcasted tensor
    # followed by element-wise subtraction
    gradInput = gradOutput - np.exp(logSoftMaxOutput) * sum

    return gradInput
```

Can you please check this out and comment?

Thanks.

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
